go-controller/pkg/node/vrfmanager/vrf_manager.go

```diff
@@ -0,0 +1,346 @@
+			klog.Errorf("Failed during LinkSubscribe callback: %v", err)
+			// Note: Not calling sync() from here: it is redundant and unsafe when stopChan is closed.
+		},
+	}
+
+	subscribe := func() (bool, chan netlink.LinkUpdate, error) {
+		linkChan := make(chan netlink.LinkUpdate)
+		if err := netlink.LinkSubscribeWithOptions(linkChan, stopCh, linkSubscribeOptions); err != nil {
+			return false, nil, err
+		}
+		// Ensure VRFs are in sync while subscribing for Link events.
+		err := vrfm.reconcile()
+		if err != nil {
+			klog.Errorf("VRF Manager: Error while reconciling VRFs, err: %v", err)
+		}
+		return true, linkChan, nil
+	}
+	return vrfm.runInternal(stopCh, doneWg, subscribe)
+}
+
+type subscribeFn func() (bool, chan netlink.LinkUpdate, error)
+
+func (vrfm *Controller) runInternal(stopChan <-chan struct{}, doneWg *sync.WaitGroup,
+	subscribe subscribeFn) error {
+	// Get the current network namespace handle
+	currentNs, err := ns.GetCurrentNS()
+	if err != nil {
+		return fmt.Errorf("error retrieving current net namespace, err: %v", err)
+	}
+	doneWg.Add(1)
+	go func() {
+		defer doneWg.Done()
+		err = currentNs.Do(func(netNS ns.NetNS) error {
+			linkSyncTimer := time.NewTicker(60 * time.Second)
+			defer linkSyncTimer.Stop()
+
+			subscribed, linkUpdateCh, err := subscribe()
+			if err != nil {
+				klog.Errorf("VRF Manager: Error during netlink subscribe for Link, err: %v", err)
+			}
+
+			for {
+				select {
+				case linkUpdateEvent, ok := <-linkUpdateCh:
+					linkSyncTimer.Reset(60 * time.Second)
+					if !ok {
+						if subscribed, linkUpdateCh, err = subscribe(); err != nil {
+							klog.Errorf("VRF Manager: Error during netlink re-subscribe due to channel closing: %v", err)
+						}
+						continue
+					}
+					ifName := linkUpdateEvent.Link.Attrs().Name
+					klog.V(3).Infof("VRF Manager: link update received for interface %s", ifName)
+					err = vrfm.syncVRF(linkUpdateEvent.Link)
+					if err != nil {
+						klog.Errorf("VRF Manager: Error syncing link %s update event, err: %v", ifName, err)
+					}
+
+				case <-linkSyncTimer.C:
+					if subscribed {
+						klog.V(5).Info("VRF Manager: calling reconcile() explicitly")
+						if err = vrfm.reconcile(); err != nil {
+							klog.Errorf("VRF Manager: Error while reconciling VRFs, err: %v", err)
+						}
+					} else {
+						if subscribed, linkUpdateCh, err = subscribe(); err != nil {
+							klog.Errorf("VRF Manager: Error during netlink re-subscribe: %v", err)
+						}
+					}
+				case <-stopChan:
+					return nil
+				}
+			}
+		})
+		if err != nil {
+			klog.Errorf("VRF Manager: failed to run link reconcile goroutine, err: %v", err)
+		}
+	}()
+	klog.Info("VRF manager is running")
+	return nil
+}
+
+func (vrfm *Controller) reconcile() error {
+	vrfm.mu.Lock()
+	defer vrfm.mu.Unlock()
+	start := time.Now()
+	defer func() {
+		klog.V(5).Infof("VRF Manager: reconciling VRFs took %v", time.Since(start))
+	}()
+
+	for _, vrf := range vrfm.vrfs {
+		if vrf.deleted {
```

Comment:
Delete doesnt depend on this field anymore for the delete track. 

May as well remove it.